pytorch_lightning/overrides/data_parallel.py

```diff
@@ -161,7 +164,30 @@ def parallel_apply(self, replicas, inputs, kwargs):
         return parallel_apply(replicas, inputs, kwargs, self.device_ids[:len(replicas)])
 
     def forward(self, *inputs, **kwargs):  # pragma: no-cover
-        self._sync_params()
+        # TODO: Update uneven inputs code path when PyTorch 1.8 is released.
+        if DDP_JOIN_AND_REBUILD_BUCKETS_AVAILABLE and self.ddp_join_enabled:
```

Comment:
Can we put this inside its own function. Maybe `self.rebuild_buckets_on_first_forward`

Reply:
We can do this, but currently PT lightning DDP is mostly copy-pasted from PT native DDP with a few modifications. I presumed we'd want to keep modifications as little as possible just so updates are as easy as possible, but I may be wrong.